### DISCUSSION TITLE
FileNotFoundError when deploying to Dagster Cloud through Cloud Interface & Github Actions

### DISCUSSION THREAD

**Discussion by Minderwinter** [dagster-io/dagster#19115] [Q&A] [answered]
I'm trying to add another code location to my dagster cloud instance from a github repo. In the Github logs I get the following error:
`FileNotFoundError: [Errno 2] No such file or directory: 
'/tmp/tmpjtudyzql/source-tmp-70d5a2ed-ddd7-4584-8168-c9f083fab937.pex'`

Not sure what could be causing this, since this tmp directory is something setup by the script itself. I've attached the relevant portion of the logs in case that's helpful.  
[dagster_deploy_error.txt](https://github.com/dagster-io/dagster/files/13880741/dagster_deploy_error.txt)

**Comment by shalabhc** [comment]
Hi @Minderwinter - can you please share your requirements.txt or setup.py (specifically the dependencies section) so we can investigate further. Thank you.

**Reply by Minderwinter** [reply]
Added below, thanks.

[requirements.txt](https://github.com/dagster-io/dagster/files/13880848/requirements.txt)

setup.py:
'    classifiers=[
        "Programming Language :: Python :: 3.8",
        "Programming Language :: Python :: 3.9",
        "Programming Language :: Python :: 3.10",
        "Operating System :: OS Independent",
    ],
    packages=find_packages(exclude=["test"]),
    package_data={"pine_dagster": ["analytics/*"]},
    install_requires=[
        "agate==1.7.1",
        "alembic==1.13.1",
        "aniso8601==9.0.1",
        "annotated-types==0.6.0",
        "anyio==4.2.0",
        "asn1crypto==1.5.1",
        "attrs==23.2.0",
        "Babel==2.14.0",
        "backoff==2.2.1",
        "certifi==2023.11.17",
        "cffi==1.16.0",
        "charset-normalizer==3.3.2",
        "click==8.1.7",
        "colorama==0.4.6",
        "coloredlogs==14.0",
        "croniter==2.0.1",
        "cryptography==41.0.7",
        "dagster==1.5.14",
        "dagster-cloud==1.5.14",
        "dagster-cloud-cli==1.5.14",
        "dagster-dbt==0.21.14",
        "dagster-graphql==1.5.14",
        "dagster-pipes==1.5.14",
        "dagster-snowflake==0.21.14",
        "dagster-snowflake-pandas==0.21.14",
        "dagster-webserver==1.5.14",
        "dbt-core==1.7.4",
        "dbt-extractor==0.5.1",
        "dbt-semantic-interfaces==0.4.3",
        "dbt-snowflake==1.7.1",
        "docstring-parser==0.15",
        "exceptiongroup==1.2.0",
        "filelock==3.13.1",
        "fsspec==2023.12.2",
        "github3.py==4.0.1",
        "gql==3.5.0",
        "graphene==3.3",
        "graphql-core==3.2.3",
        "graphql-relay==3.2.0",
        "grpcio==1.60.0",
        "grpcio-health-checking==1.60.0",
        "h11==0.14.0",
        "httptools==0.6.1",
        "humanfriendly==10.0",
        "idna==3.6",
        "importlib-metadata==6.11.0",
        "importlib-resources==6.1.1",
        "isodate==0.6.1",
        "jaraco.classes==3.3.0",
        "Jinja2==3.1.2",
        "jsonschema==4.20.0",
        "jsonschema-specifications==2023.12.1",
        "keyring==24.3.0",
        "leather==0.3.4",
        "Logbook==1.5.3",
        "Mako==1.3.0",
        "markdown-it-py==3.0.0",
        "MarkupSafe==2.1.3",
        "mashumaro==3.11",
        "mdurl==0.1.2",
        "minimal-snowplow-tracker==0.0.2",
        "more-itertools==10.2.0",
        "msgpack==1.0.7",
        "multidict==6.0.4",
        "networkx==3.1",
        "numpy==1.24.4",
        "orjson==3.9.10",
        "packaging==23.2",
        "pandas==2.0.3",
        "parsedatetime==2.6",
        "pathspec==0.11.2",
        "pendulum==2.1.2",
        "pex==2.1.156",
        "pkgutil_resolve_name==1.3.10",
        "platformdirs==3.11.0",
        "prompt-toolkit==3.0.36",
        "protobuf==4.25.1",
        "pyarrow==10.0.1",
        "pycparser==2.21",
        "pydantic==2.5.3",
        "pydantic_core==2.14.6",
        "Pygments==2.17.2",
        "PyJWT==2.8.0",
        "pyOpenSSL==23.3.0",
        "python-dateutil==2.8.2",
        "python-dotenv==1.0.0",
        "python-slugify==8.0.1",
        "pytimeparse==1.1.8",
        "pytz==2023.3.post1",
        "pytzdata==2020.1",
        "PyYAML==6.0.1",
        "questionary==2.0.1",
        "referencing==0.32.1",
        "requests==2.31.0",
        "requests-toolbelt==1.0.0",
        "rich==13.7.0",
        "rpds-py==0.16.2",
        "shellingham==1.5.4",
        "six==1.16.0",
        "sniffio==1.3.0",
        "snowflake-connector-python==3.4.1",
        "snowflake-sqlalchemy==1.5.1",
        "sortedcontainers==2.4.0",
        "SQLAlchemy==1.4.51",
        "sqlparse==0.4.4",
        "starlette==0.34.0",
        "tabulate==0.9.0",
        "text-unidecode==1.3",
        "tomli==2.0.1",
        "tomlkit==0.12.3",
        "toposort==1.10",
        "tqdm==4.66.1",
        "typer==0.9.0",
        "typing_extensions==4.9.0",
        "tzdata==2023.4",
        "universal-pathlib==0.1.4",
        "uritemplate==4.1.1",
        "urllib3==1.26.18",
        "uvicorn==0.25.0",
        "uvloop==0.19.0",
        "watchdog==3.0.0",
        "watchfiles==0.21.0",
        "wcwidth==0.2.13",
        "websockets==12.0",
        "yarl==1.9.4",
        "zipp==3.17.0"
    ],
    extras_require={"dev": ["dagster-webserver", "pytest"], "tests": ["mypy", "pylint", "pytest"]},'

**Reply by shalabhc** [reply]
Thank you. I am unable to reproduce the build failure with these dependencies. Can you tell me which PYTHON_VERSION is specified in your deploy.yml?

**Reply by Minderwinter** [reply]
3.8

**Reply by shalabhc** [reply]
Can you please retry the deploy to see if that works?

Also can you see if the deploy works when you remove the `package_data={"pine_dagster": ["analytics/*"]},` line? It may eventually fail to deploy but it will show if progresses beyond the build itself.

**Reply by Minderwinter** [reply]
Yep, will give it a try now. Thanks

**Reply by Minderwinter** [reply]
Unfortunately, I still got the same error.

**Reply by shalabhc** [reply]
Investigating another similar issue, we found a problem when there are invalid symlinks - ie symlinks that dont point to a real file - in the project. Do you have any such symlinks? We're still looking into why this fails in github actions, but if these symlinks are unnecessary (eg they are in a venv) then you should be able to delete them and the build should succeed.

**Reply by Minderwinter** [reply]
Hmmm, not sure. I'll check on that

**Comment by Minderwinter** [comment, marked as answer]
That did it! I did actually have a venv folder in the repo with some symlinks pointing outside. Removed the venv and I got through the build. Thank you!
